Source/WebGPU/WGSL/TypeCheck.cpp

```diff
@@ -79,12 +85,14 @@ class TypeChecker : public AST::Visitor, public ContextProvider<Type*> {
     Type* m_u32;
     Type* m_f32;
     FixedVector<TypeConstructor> m_typeConstrutors;
+    WTF::Vector<Error> m_errors;
```

Comment:
I kind of think `Error` and `m_errors` are bad names, because warnings go here too.

Reply:
`Error` and `Warning` are aliases to the same underlying class (`CompilationMessage`). I think we should probably keep them separate, no? It would be helpful to know whether there were any errors or not, although that would mess up the order of reporting... either way, we're only reporting errors for now, I can rename it once we get to warnings.